### DISCUSSION TITLE
Support permalinks for self-hosted Gitlab instances

### DISCUSSION THREAD

**Discussion by collinmurch** [zed-industries/zed#19860] [Feature Requests]
Hey hey, Zed fan here.

I've got a pretty minor issue that I've come across with Zed specifically when using it at work. One of the workflows I use most frequently in VSCode and similar tools is the "copy permalink" feature, which I'm pleased that Zed supports.

Unfortunately my use-case, it seems that this is limited to repositories hosted on the official servers (`gitlab.com`). The company I work for uses a self-hosted instance which causes all permalink generation requests to fail as the remote urls look like `git@gitlab.[DOMAIN].[com/net/etc.]` which is not matched [here](https://github.com/zed-industries/zed/blob/93b20008e04a73b55f0448c508e9b1ccda713abd/crates/git_hosting_providers/src/providers/gitlab.rs#L29C19-L29C20), causing an error.

I'm happy to implement the change but want to know if:
- You folks are okay with expanding the matching criteria for Gitlab
- Or if I should add a "fallback" matching for these self-hosted instances for other providers (I'm as familiar with self-hosting on non-Gitlab hosting services like Bitbucket etc.)

Thanks!

**Comment by opsnull** [comment]
thank you, please go ahead.
https://github.com/zed-industries/zed/issues/19735

**Reply by collinmurch** [reply]
https://github.com/zed-industries/zed/pull/19868, let me know if there's anything else needed.

**Reply by collinmurch** [reply]
Looks like this is a duplicate of the other discussion, will move to there. My bad, missed it.
